deployments/cloudfoundry/buildpack/bin/supply

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -euo pipefail
+
+BUILD_DIR=$1
+CACHE_DIR=$2
+DEPS_DIR=$3
+IDX=$4
+
+BUILDPACK_DIR="$(readlink -f ${BASH_SOURCE%/*})"
+TARGET_DIR="$BUILD_DIR/.otelcollector"
+
+OTEL_CONFIG="${OTEL_CONFIG:-otelconfig.yaml}"
+# Version must be >=0.48.0, the first release included the enabled Cloud Foundry receiver
+OTEL_VERSION="${OTEL_VERSION:-latest}"
+# Cloud Foundry only supports Linux stacks currently, but if that changes this variable can be modified
+# for the proper stack.
+OS="${OS:-linux_amd64}"
+OTEL_BASE_URL="https://github.com/signalfx/splunk-otel-collector/releases"
+
+echo "-----> Installing Otel Collector ${OTEL_VERSION}"
+echo "       BUILD_DIR: $BUILD_DIR"
+echo "       CACHE_DIR: $CACHE_DIR"
+echo "       DEPS_DIR: $DEPS_DIR"
+echo "       BUILDPACK_INDEX: $IDX"
+echo "       BUILDPACK_DIR: $BUILDPACK_DIR"
+echo "       TARGET_DIR: $TARGET_DIR"
+
+mkdir -p $TARGET_DIR
+
+# Copy over default OpenTelemetry configuration file for sidecar configuration use
+cp "$BUILDPACK_DIR/../$OTEL_CONFIG" "$TARGET_DIR/"
+
+# Get proper version number if we're getting latest release
+if [ $OTEL_VERSION = "latest" ] && [ -z "${OTEL_BINARY_DOWNLOAD_URL:-}" ]; then
+        OTEL_VERSION=$( wget -qO- --header="Accept: application/json" "$OTEL_BASE_URL/latest" | jq -r '.tag_name' )
+        if [ -z "$OTEL_VERSION" ]; then
+            echo "Failed to get tag_name for latest release from $OTEL_BASE_URL/latest" >&2
+            exit 1
+        fi
+    fi
+
+# File name for given release version
+OTEL_BINARY="${OTEL_BINARY:-otelcontribcol_${OS}-${OTEL_VERSION}}"
+CACHED_OTEL_BINARY="$CACHE_DIR/$OTEL_BINARY"
+
+if [[ -f "$CACHED_OTEL_BINARY" ]]; then
+  echo "-----> Using cached Otel Collector install: $CACHED_OTEL_BINARY"
+  cp $CACHED_OTEL_BINARY $TARGET_DIR
+else
+  OTEL_BINARY_DOWNLOAD_URL="${OTEL_BINARY_DOWNLOAD_URL:-${OTEL_BASE_URL}/download/${OTEL_VERSION}/otelcol_${OS}}"
+  echo "-----> Downloading OpenTelemetry Collector $OTEL_VERSION ($OTEL_BINARY_DOWNLOAD_URL)"
+  wget -nv -O "$TARGET_DIR/$OTEL_BINARY" $OTEL_BINARY_DOWNLOAD_URL
```

Comment:
Can you also download and patch the smart agent bundle like the dockerfile does? https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/Dockerfile#L27 `SPLUNK_BUNDLE_DIR` and `SPLUNK_COLLECTD_DIR` env vars should also be set for these to be detected.

Reply:
I'm not very familiar with the smart agent, or with the SignalFx receiver, so I just have some clarifying questions:

1. From your description, this requirement is because of the SignalFx receiver, so is this executable bundle required with every collector that wants to use the SignalFx receiver?
2. This is specifically a SignalFx receiver dependency, and not a Smart Agent dependency right? I originally assumed this functionality was relevant only for the Smart Agent, so I want to make sure that my assumption was wrong here.

Reply:
I talked to Ryan so I'll just include answers here.
1. This is required for every collector that wants to use the [SignalFx SmartAgent receiver](https://github.com/signalfx/splunk-otel-collector/blob/main/internal/receiver/smartagentreceiver/README.md#signalfx-smart-agent-receiver). The Smart Agent receiver is a way to allow use of all Smart Agent monitors, so the bundle is really just the release of the Smart Agent.
2. This is a Smart Agent dependency, but since the receiver relies on the Smart Agent, it also relies on the bundle as well.
Note: This is regarding the smartagent receiver, not the signalfx receiver.